packages/langium/src/generator/generator-node.ts

```diff
@@ -214,8 +214,8 @@ export class CompositeGeneratorNode {
      *       '}'
      *   );
      */
-    indent(configurator?: (indentingNode: IndentNode) => void, indentation?: string|number): this {
-        const node = new IndentNode(indentation, false);
+    indent(configurator?: (indentingNode: IndentNode) => void, indentation?: string|number, indentImmediately = true): this {
```

Comment:
What's the use case of the configurator function? Could the `indentImmediately` property be configured with that?

I like to avoid Boolean function arguments. Alternative signature (backwards-compatible):
```
indent(configurator?: (indentingNode: IndentNode) => void, options?: string | number | { indentation?: string | number,  indentImmediately: boolean })
```

Reply:
The `configurator` is a callback allowing to use `indent` in a fluent api style, see the example of the jsdoc:
https://github.com/langium/langium/blob/d53d019cd3f2842e5596f78fa1733e0050f4b7a7/packages/langium/src/generator/generator-node.ts#L205-L213

What's the problem with boolean args? Readability?

I expect `indentImmediately` to be omitted by client code in 99% of the use cases, and I don't expect any further things to be added there. In the initial draft I hoped I could avoid that parameter, it need to be `false` when creating indent nodes via `expandToNode`. That's why I hard-coded it to `false`. However, that breaks the above example, as the first line would not be indented as expected.

Reply:
> What's the problem with boolean args? Readability?

Yes, calling `myFunction(false)` is more obscure than `myFunction({ indentImmediately: false })`

Let's leave that method as you suggested, we can still add an object-based alternative  signature later.

Reply:
> I expect indentImmediately to be omitted by client code in 99% of the use cases

Rethinking about this, an argument that is expected to be rarely used is even more suited to be wrapped in an object _(named argument)_.